File: src/org/cappuccino_project/ide/intellij/plugin/stubs/impl/ObjJClassDeclarationStubImpl.java
package org.cappuccino_project.ide.intellij.plugin.stubs.impl;

import com.intellij.psi.stubs.IStubElementType;
import com.intellij.psi.stubs.StubElement;
import com.intellij.util.io.StringRef;
import org.cappuccino_project.ide.intellij.plugin.psi.interfaces.ObjJClassDeclarationElement;
import org.cappuccino_project.ide.intellij.plugin.stubs.interfaces.ObjJClassDeclarationStub;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public class ObjJClassDeclarationStubImpl<PsiT extends ObjJClassDeclarationElement<? extends ObjJClassDeclarationStub>> extends ObjJStubBaseImpl<PsiT> implements ObjJClassDeclarationStub<PsiT> {
    private final String className;
    private final List<String> inheritedProtocols;

    ObjJClassDeclarationStubImpl(
            @NotNull
                    StubElement parent,
            @NotNull
                    IStubElementType elementType,
            @NotNull
            final String className,
            @NotNull
            final List<String> inheritedProtocols) {
        super(parent, elementType);
        this.className = className;
        this.inheritedProtocols = inheritedProtocols;
    }

    @NotNull
    @Override
    public String getClassName() {
        return this.className;
    }

    @NotNull
    @Override
    public List<String> getInheritedProtocols() {
        return this.inheritedProtocols;
    }
}

File: src/cappuccino/ide/intellij/plugin/stubs/impl/ObjJClassDeclarationStubImpl.kt
package cappuccino.ide.intellij.plugin.stubs.impl

import cappuccino.ide.intellij.plugin.psi.interfaces.ObjJClassDeclarationElement
import cappuccino.ide.intellij.plugin.stubs.interfaces.ObjJClassDeclarationStub
import com.intellij.psi.stubs.IStubElementType
import com.intellij.psi.stubs.StubElement

open class ObjJClassDeclarationStubImpl<PsiT : ObjJClassDeclarationElement<out ObjJClassDeclarationStub<*>>> internal constructor(
        parent: StubElement<*>,
        elementType: IStubElementType<*, *>,
        override val className: String,
        override val inheritedProtocols: List<String>,
        private val shouldResolve: Boolean) : ObjJStubBaseImpl<PsiT>(parent, elementType), ObjJClassDeclarationStub<PsiT> {

    override fun shouldResolve(): Boolean {
        return shouldResolve
    }
}
